include/fc/io/cfile.hpp

```diff
@@ -188,13 +227,21 @@ class datastream<fc::cfile, void> : public fc::cfile {
 
    bool get(char& c) {
       c = this->getc();
+      _current_c = c;
       return true;
    }
 
-   bool remaining() { return !this->eof(); }
+   bool remaining() { return (EOF != _current_c); }
```

Comment:
This still has essentially the same problem as the original.  It checks whether you attempted to read past the end, not whether the datastream is at the end.

Reply:
Thank you  for  pointing that out. I addressed this issue,  now.